app/lib/backend_api_logic/service_extension.rb

```diff
@@ -40,5 +42,37 @@ def create_first_backend_api_config!
       attrs = {backend_api: backend_api, path: ''}
       backend_api_configs.public_send(constructor, attrs)
     end
+
+    class BackendApiProxy
+      attr_reader :service
+      delegate :account, :backend_apis, :backend_api_configs, to: :service
+
+      def initialize(service)
+        @service = service
+      end
+
+      def backend_api_config
+        @backend_api_config ||= backend_api.backend_api_configs.first ||
```

Comment:
This should do it, no?

```suggestion
        @backend_api_config ||= backend_api_configs.first ||
```

If `Service` doesn't have any `BackendApiConfig`, then `#backend_api` will do `account.backend_apis.build`. Yet `backend_api.backend_api_configs.first` is `nil`, so the fallback invokes `#backend_api` again and builds a first `BackendApiConfig` using the `BackendApi` just built.

If `Service` already has a `BackendApiConfig`, then `#backend_api` will return the `BackendApi` of the first `BackendApiConfig` of the `Service` (let's call it `service_a`). However, the **first** `BackendApiConfig` of this returned `BackendApi` may not be the one with `service_a`, but, for example, with another `Service` called `service_b`; ergo `service_a.backend_api_proxy.backend_api_config.service == service_b`. 🤔

Reply:
Good catch! thank you!